### PR TITLE
Fix handling for non-existent or empty config files

### PR DESCRIPTION
**Pull request by JakeGuy11**
Foreword: I'm not very experienced with pull requests so I don't know if I did it right. If not, please let me know and I'll fix it so it can be merged!\
Also, I'm directly copying this from my pull request from the master since it's unmaintained.

I recently noticed that if I manually remove the contents of `token.yml` but keep the file, the program throws an error. A little playing around and I found that the same things happen with entries in other config files.

These changes fix the handling for that. Instead of throwing errors, it will warn the user that their entries are invalid and manually set them to the default.

I considering a few other things in place of hard coding in the default config values (in case they ever change), but in the end, I decided not to. My main alternative would have been to have default `cli-defaults.yml`, `bin-path.yml` and `dir-path.yml` files somewhere else that it would pull from instead. If the maintainers think that's a better idea, let me know and I can do that.

### CODE REVIEW

**Comment by izu-co** [comment]
It’s pretty late here so you might have to wait a day or so for my review. One thing I can see right away is that there conflicted (meaning I made changed to the same line or so). Can you resolve the conflicts please?

**Comment by JakeGuy11** [comment]
Yeah, no rush. I'll see what I can do!

**Review by izu-co — CHANGES_REQUESTED** [review]
I would approve of the defaults for the `cfg.dir` as well as the changes to the token file.
The rest is just optional and does not need a default. If you set the defaults for the CLI, for example, I couldn't delete the options if I want to use the defaults from yargs.

Also, please use `Object.prototype.hasOwnProperty.call()` instead of `hasOwnPrperty`. I added an ESLint check, since the style is required from here on out.

So, to sum it up, check that ESLint has no errors and remove the defaults for the `bin` and `cli` (the warnings are OK tho), please.

**Comment by JakeGuy11** [comment]
Should be better now - removed the defaults for `bin` and `cli`, and using the eslint config from this repo gave no errors.

**Review by izu-co — APPROVED** [review]
Thanks for the changes.

**Comment by izu-co** [comment]
The changes will be included in the next release, since they are no bug fixes.
